examples/hello_compute_c/main.c

```diff
@@ -0,0 +1,135 @@
+        });
+
+    WGPUDeviceId device = wgpu_adapter_request_device(adapter,
+        &(WGPUDeviceDescriptor){
+            .extensions =
+                {
+                    .anisotropic_filtering = false,
+                },
+        });
+
+    WGPUBufferId staging_buffer = wgpu_device_create_buffer_mapped(device,
+        &(WGPUBufferDescriptor){.size = numbers_length,
+            .usage = WGPUBufferUsage_MAP_READ | WGPUBufferUsage_TRANSFER_DST |
+                WGPUBufferUsage_TRANSFER_SRC},
+        numbers);
+
+    WGPUBufferId storage_buffer = wgpu_device_create_buffer(device,
+        &(WGPUBufferDescriptor){.size = size,
+            .usage = WGPUBufferUsage_STORAGE | WGPUBufferUsage_TRANSFER_DST |
+                WGPUBufferUsage_TRANSFER_SRC});
+
+    WGPUBindGroupLayoutId bind_group_layout =
+        wgpu_device_create_bind_group_layout(device,
+            &(WGPUBindGroupLayoutDescriptor){
+                .bindings = &(WGPUBindGroupLayoutBinding){.binding = 0,
+                    .visibility = WGPUShaderStage_COMPUTE,
+                    .ty = WGPUBindingType_StorageBuffer}});
+
+    WGPUBindGroupId bind_group = wgpu_device_create_bind_group(device,
+        &(WGPUBindGroupDescriptor){.layout = bind_group_layout,
+            .bindings = &(WGPUBindGroupBinding){
+				.binding = 0
+	}});
+
+
+    WGPUPipelineLayoutId pipeline_layout =
+        wgpu_device_create_pipeline_layout(device,
+            &(WGPUPipelineLayoutDescriptor){
+                .bind_group_layouts = &bind_group_layout});
+
+    WGPUShaderModuleId shader_module = wgpu_device_create_shader_module(device,
+        &(WGPUShaderModuleDescriptor){
+            .code = read_file("./../../data/shader.comp.spv")});
+
+	WGPUComputePipelineId compute_pipeline =
+			wgpu_device_create_compute_pipeline(device,
+				&(WGPUComputePipelineDescriptor){
+					.layout = pipeline_layout,
+					.compute_stage = &(WGPUPipelineStageDescriptor){
+					.module = shader_module, 
+					.entry_point = "main"}});
+
+    WGPUCommandEncoderId encoder = wgpu_device_create_command_encoder(
+        device, &(WGPUCommandEncoderDescriptor){.todo = 0});
+
+    wgpu_command_buffer_copy_buffer_to_buffer(
+        encoder, staging_buffer, 0, storage_buffer, 0, size);
+
+    WGPUComputePassId command_pass =
+        wgpu_command_encoder_begin_compute_pass(encoder);
+    wgpu_compute_pass_set_pipeline(command_pass, compute_pipeline);
+
+    wgpu_compute_pass_set_bind_group(command_pass, 0, bind_group, 0, 0);
+    wgpu_compute_pass_dispatch(command_pass, numbers_length, 1, 1);
+
+	wgpu_command_buffer_copy_buffer_to_buffer(
+        encoder, storage_buffer, 0, staging_buffer, 0, size);
+
+	WGPUQueueId queue = wgpu_device_get_queue(device);	
+
+	WGPUCommandBufferId command_buffer =
+            wgpu_command_encoder_finish(encoder);
+
+	wgpu_queue_submit(queue, &command_buffer, 1);
+
+	wgpu_buffer_map_read_async(staging_buffer, 0, size, &ReadBufferMap, 1);
```

Comment:
The `1` here should be `uint8_t *userdata`, but it looks like we don't use user data in `ReadBufferMap`, so we can just pass `NULL` here